json-rpc/src/methods.rs

```diff
@@ -302,11 +304,7 @@ async fn get_account(
 /// Can be used to verify that target Full Node is up-to-date
 async fn get_metadata(service: JsonRpcService, request: JsonRpcRequest) -> Result<MetadataView> {
     let chain_id = service.chain_id().id();
-    let version = if !request.params.is_empty() {
-        request.parse_version_param(0, "version")?
-    } else {
-        request.version()
-    };
+    let version = request.parse_version_param(0, "version")?;
```

Comment:
I'm assuming that this logic happens in this function (handling the case where its not provided?)

Reply:
yes, it handles the case given parameter index is > len(request params)